WalletWasabi/WabiSabi/Client/CoinJoinResult.cs

```diff
@@ -0,0 +1,18 @@
+using NBitcoin;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using WalletWasabi.Blockchain.TransactionOutputs;
+
+namespace WalletWasabi.WabiSabi.Client;
+
+public record CoinJoinResult(
```

Comment:
This goes in the right direction but the class contains everything needed to communicate the success of a coinjoin and a boolean flag in case it failed, that's what I thing should be improved.

I think we should have a `CoinJoinResult` record cointaining general info like `RoundId` and subclases `CoinJoinSucessfulResult` with the inputs/outputs, etc and a `CoinJoinFailureResult` with the reason (`RoundDisrupted/GoToBlameRound`, `WrongPhase/RoundEnded`, etc)

Reply:
That is causing confusion in me. In an unrecoverable error, the client will throw an exception. In other cases we will get the result. There cannot be RoundEnded field in the CoinJoinResult because in that case, the client will throw.